lib/SILOptimizer/Transforms/SpecializeOpaqueArchetypes.cpp

```diff
@@ -394,24 +405,7 @@ void OpaqueSpecializerCloner::insertOpaqueToConcreteAddressCasts(
       auto argIdx = apply.getCalleeArgIndex(opd);
       auto argType = substConv.getSILArgumentType(argIdx);
       if (argType.getASTType() != opd.get()->getType().getASTType()) {
-        if (argConv.isIndirectConvention()) {
-          auto cast = getBuilder().createUncheckedAddrCast(apply.getLoc(),
-                                                           opd.get(), argType);
-          opd.set(cast);
-        } else if (argType.is<SILFunctionType>()) {
-          auto cast = getBuilder().createConvertFunction(
-              apply.getLoc(), opd.get(), argType,
-              /*withoutActuallyEscaping*/ false);
-          opd.set(cast);
-        } else if (argType.isTrivial(getBuilder().getFunction())) {
-          auto cast = getBuilder().createUncheckedTrivialBitCast(
-              apply.getLoc(), opd.get(), argType);
-          opd.set(cast);
-        } else {
-          auto cast = getBuilder().createUncheckedRefCast(apply.getLoc(),
-                                                          opd.get(), argType);
-          opd.set(cast);
-        }
+        opd.set(createCast(apply.getLoc(), opd.get(), argType));
```

Comment:
Is createCast() equivalent to castValueToABICompatibleType() in include/swift/SILOptimizer/Utils/Local.h?

Reply:
Yeah looks like it might be. I will fix in a follow-up.

Reply:
Thank you!

Reply:
Follow up: I can not use castValueToABICompatibleType because I does only handle type aggregates (no structs).